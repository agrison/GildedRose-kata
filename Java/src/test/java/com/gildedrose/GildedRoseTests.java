package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

public class GildedRoseTests {
    @Test
    public void updateQuality() {
        CombinationApprovals.verifyAllCombinations(
                this::callUpdateQuality,
                new String[]{"a common item"},
                new Integer[]{0},
                new Integer[]{0}
        );
    }

    private String callUpdateQuality(String name, int sellIn, int quality) {
        var items = new Item[]{new Item(name, sellIn, quality)};
        var gildedRose = new GildedRose(items);
        gildedRose.updateQuality();

        return gildedRose.items[0].toString();
    }
}
