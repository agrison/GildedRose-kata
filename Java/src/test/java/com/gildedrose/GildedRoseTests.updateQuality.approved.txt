[a common item, 0, 0] => a common item, -1, 0 
